src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsMPM.cpp

```diff
@@ -0,0 +1,897 @@
+    setApplyDefaultValue( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Flag to indicate the use of the incremental displacement from the previous step as an "
+                    "initial estimate for the incremental displacement of the current step." );
+
+  registerWrapper( viewKeyStruct::maxNumResolvesString(), &m_maxNumResolves ).
+    setApplyDefaultValue( 10 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Value to indicate how many resolves may be executed after some other event is executed. "
+                    "For example, if a SurfaceGenerator is specified, it will be executed after the mechanics solve. "
+                    "However if a new surface is generated, then the mechanics solve must be executed again due to the "
+                    "change in topology." );
+
+  registerWrapper( viewKeyStruct::strainTheoryString(), &m_strainTheory ).
+    setApplyDefaultValue( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Indicates whether or not to use "
+                    "`Infinitesimal Strain Theory <https://en.wikipedia.org/wiki/Infinitesimal_strain_theory>`_, or "
+                    "`Finite Strain Theory <https://en.wikipedia.org/wiki/Finite_strain_theory>`_. Valid Inputs are:\n"
+                    " 0 - Infinitesimal Strain \n"
+                    " 1 - Finite Strain" );
+
+  registerWrapper( viewKeyStruct::contactRelationNameString(), &m_contactRelationName ).
+    setApplyDefaultValue( viewKeyStruct::noContactRelationNameString() ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Name of contact relation to enforce constraints on fracture boundary." );
+
+  registerWrapper( viewKeyStruct::maxForceString(), &m_maxForce ).
+    setInputFlag( InputFlags::FALSE ).
+    setDescription( "The maximum force contribution in the problem domain." );
+
+}
+
+void SolidMechanicsMPM::postProcessInput()
+{
+  SolverBase::postProcessInput();
+
+//  checkModelNames( m_solidMaterialNames, viewKeyStruct::solidMaterialNamesString() );
+
+  LinearSolverParameters & linParams = m_linearSolverParameters.get();
+  linParams.isSymmetric = true;
+  linParams.dofsPerNode = 3;
+  linParams.amg.separateComponents = true;
+}
+
+SolidMechanicsMPM::~SolidMechanicsMPM()
+{
+  // TODO Auto-generated destructor stub
+}
+
+
+void SolidMechanicsMPM::registerDataOnMesh( Group & meshBodies )
+{
+  ExecutableGroup::registerDataOnMesh( meshBodies );
+
+  forMeshTargets( meshBodies, [&] ( string const & meshBodyName,
+                                    MeshLevel & meshLevel,
+                                    arrayView1d< string const > const & regionNames )
+  {
+    ParticleManager & particleManager = meshLevel.getParticleManager();
+
+    MeshBody const & meshBody = meshBodies.getGroup< MeshBody >( meshBodyName );
+
+    // Set constitutive names on particles
+    if( meshBody.hasParticles() )
```

Comment:
I do not know your business, put is it legit to continue the simulation if you have no particle (maybe yes, just asking.)

Reply:
It is common for there to me no particles within a single partition for some or all of a simulation.  Particles move through the grid and the entire domain is uniformly partitioned spatially.